router/router.go

```diff
@@ -1760,7 +1770,56 @@ func (rt *HandleT) generatorLoop(ctx context.Context) {
 	}
 }
 
-func (rt *HandleT) readAndProcess() int {
+func (rt *HandleT) getValueForParameter(destination backendconfig.DestinationT, parameter string) string {
+	switch {
+	case parameter == "destination_id":
+		return destination.ID
+	default:
+		panic(fmt.Errorf("rt: %s: Unknown parameter(%s) to find value from router destination %+v", rt.destName, parameter, destination))
+	}
+}
+
+func (rt *HandleT) constructParameterFilters(batchDest backendconfig.DestinationT) []jobsdb.ParameterFilterT {
+	parameterFilters := make([]jobsdb.ParameterFilterT, 0)
+	for _, key := range rt.queryFilters.ParameterFilters {
+		parameterFilter := jobsdb.ParameterFilterT{
+			Name:     key,
+			Value:    rt.getValueForParameter(batchDest, key),
+			Optional: false,
+		}
+		parameterFilters = append(parameterFilters, parameterFilter)
+	}
+
+	return parameterFilters
+}
```

Comment:
can't we drop `rt.queryFilters` & `getValueForParameter` altogether and simplify this further?
Even better we can skip this method too and construct `parameterFilters` directly in `getQueryParamsForDest` method that was proposed before.

```suggestion
func (rt *HandleT) getParameterFilters(destination backendconfig.DestinationT) []jobsdb.ParameterFilterT {
	return []jobsdb.ParameterFilterT{{
		Name:     "destination_id",
		Value:    destination.ID,
		Optional: false,
	}}
}
```